### ISSUE TITLE
OAuth in app web view

### ISSUE THREAD

**Issue by macsterisk** [mattdonnelly/Swifter#104]
Is there a way to specify a web view to be used to do authentication rather than switching to Safari?  I don't want the user to have to leave my app.  Thanks!

**Comment by alexdd55** [comment]
apple will complain about leaving the app anyway, so you need to do it in-app.

i added https://github.com/dongri/OAuthSwift and did this
``` extension OAuthSwiftOpenURLExternally {
	@objc public func handle(url: NSURL) {
		NSNotificationCenter.defaultCenter().postNotificationName("AuthNotifier", object: url)
	}
}
```
and used the OAuthSwift apllication-openurl code.
this worked with swift 1.2

therefore that i am not finished converting my whole app, i can't say if it is still working...

**Comment by alexdd55** [comment]
UPDATE:
in the new version its all shown in the demo:
i think @meteochu  committed it here:
https://github.com/mattdonnelly/Swifter/blob/master/SwifterDemoiOS/AuthViewController.swift

might be possible that you have to make navigation adjustement in line 85 to line 87, if SFSafariController is not available for you

**Comment by meteochu** [comment]
Yeah, I added that in a while ago when I needed it. I still need to fix the parameters in `authorizeWithCallbackURL:` since it's a bit messy. It works as is, but can definitely be improved.

To check if SFSafariViewController is available or not, simply check with 
```if #available (iOS 9.0, *) {
    // SFSafariViewController Code
} else {
    // Fallback on previous iOS versions. Use UIApplication.sharedApplication().openURL(url)
}
```

Using SFSafariViewController should be the same as using the old method so you do need to have `Swifter.handleOpenURL(url)` within the AppDelegate

**Comment by alexdd55** [comment]
@meteochu just working on converting, but will get back to SFSafariController asap. Thanks for the fix :+1:
